syntax = "proto3";

package signal;


message Complex {
    double real = 1;
    double imag = 2;
}


message RuningSumRequest {
    repeated Complex signal = 1;      
}


message RuningSumResponse {
    repeated Complex result = 1; 
}


service SignalService {
    rpc ComputeRuningSum (RuningSumRequest) returns (RuningSumResponse);
}